python-sdk/nuscenes/lidarseg/class_histogram.py

```diff
@@ -99,57 +117,50 @@ def render_lidarseg_histogram(nusc: NuScenes,
-
-    # Format the x-axis.
-    ax.set_xlabel(x_label, fontsize=font_size)
-    ax.set_xticklabels(class_names, rotation=45, horizontalalignment='right',
-                       fontweight='light', fontsize=font_size)
-
-    # Shift the class names on the x-axis slightly to the right for aesthetics.
-    trans = mtrans.Affine2D().translate(10, 0)
-    for t in ax.get_xticklabels():
-        t.set_transform(t.get_transform() + trans)
-
-    # Format the y-axis.
-    ax.set_ylabel(y_label, fontsize=font_size)
-    ax.set_yticklabels(counts, size=font_size)
-
-    # Transform the y-axis to log scale.
-    if y_log_scale:
-        ax.set_yscale("log")
-
-    # Display the y-axis using nice scientific notation.
-    formatter = ScalarFormatter(useOffset=False, useMathText=True)
-    ax.yaxis.set_major_formatter(
-        FuncFormatter(lambda x, pos: "${}$".format(formatter._formatSciNotation('%1.10e' % x))))
-
-    if chart_title:
-        ax.set_title(chart_title, fontsize=font_size)
+    fig, axes = plt.subplots(nrows=2, sharex=True, figsize=(16, 9))
+    for ax in axes:
+        ax.margins(x=0.005)  # Add some padding to the left and right limits of the x-axis for aesthetics.
+        ax.set_axisbelow(True)  # Ensure that axis ticks and gridlines will be below all other ploy elements.
+        ax.yaxis.grid(color='white', linewidth=2)  # Show horizontal gridlines.
+        ax.set_facecolor('#eaeaf2')  # Set background of plot.
+        ax.spines['top'].set_visible(False)  # Remove top border of plot.
+        ax.spines['right'].set_visible(False)  # Remove right border of plot.
+        ax.spines['bottom'].set_visible(False)  # Remove bottom border of plot.
+        ax.spines['left'].set_visible(False)  # Remove left border of plot.
+
+    # Plot the histograms.
+    for i, (histogram, config) in enumerate(histograms_config.items()):
+        axes[i].bar(class_names, config['y_values'], color=colors)
+        assert len(class_names) == len(axes[i].get_xticks()), \
+            'There are {} classes, but {} are shown on the x-axis'.format(len(class_names), len(axes[i].get_xticks()))
+
+        # Format the x-axis.
+        axes[i].set_xticklabels(class_names, rotation=45, horizontalalignment='right',
+                                fontweight='light', fontsize=font_size)
+
+        # Shift the class names on the x-axis slightly to the right for aesthetics.
+        trans = mtrans.Affine2D().translate(10, 0)
+        for t in axes[i].get_xticklabels():
+            t.set_transform(t.get_transform() + trans)
+
+        # Format the y-axis.
+        axes[i].set_ylabel(config['y_label'], fontsize=font_size)
+        axes[i].set_yticklabels(config['y_values'], size=font_size)
+        axes[i].set_yscale(config['y_scale'])
+
+        if config['y_scale'] == 'linear':
+            axes[i].yaxis.set_major_formatter(mticker.FormatStrFormatter('%.1e'))
+            # axes[i].yaxis.set_major_formatter(MathTextSciFormatter('%1.0e'))
```

Comment:
minor: removed unused comment.

Reply:
Addressed in https://github.com/nutonomy/nuscenes-devkit/pull/658/commits/5f6250661a160adc9d304f7122bf8b88d84a7090